### PR TITLE
Program startup options: Rename options::Flags::Hidden to options::Flags::Uncommon

### PR DESCRIPTION
**Pull request by Simran-B**
### Scope & Purpose

Together with extended code comments, this should make it clearer what the real purpose of hidden options is (not listed by --help). The unfortunate naming caused troubled multiple times.

- [x] :hammer: Refactoring/simplification

### Checklist

There should be no functional change.

- [ ] :book: CHANGELOG entry made
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

https://arangodb.slack.com/archives/C1PJT16QP/p1637157654444500

- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/863

### CODE REVIEW

**Comment by Simran-B** [comment]
Already approved by @jsteemann in #15588.

**Review by gnusi — CHANGES_REQUESTED** [review]
This change makes backporting much harder without any good reason. Please close the PR.

**Comment by jsteemann** [comment]
@gnusi : I don't think the change in this PR is overly complex. So anyone running into a merge conflict/compile error when backporting should be able to figure the difference out easily. So I won't object to merging this PR.

**Comment by Simran-B** [comment]
@gnusi Changing `Hidden` to `Uncommon` should be a matter of seconds, whereas the current naming is responsible for recurring confusion and the loss of several hours of working time:
- https://arangodb.slack.com/archives/C1PJT16QP/p1637157654444500 (https://github.com/arangodb/docs/pull/818)
- https://arangodb.slack.com/archives/C1PJT16QP/p1624354482172800
- https://arangodb.slack.com/archives/C1PJT16QP/p1581520858155600?thread_ts=1581519984.148900&cid=C1PJT16QP
- https://arangodb.slack.com/archives/C1PJT16QP/p1560508774278400
- https://arangodb.slack.com/archives/C1PJT16QP/p1542639461466200
- https://arangodb.slack.com/archives/C1PJT16QP/p1528202734000392

There's no need to backport this change to 3.7-3.9, if that helps.

**Review by cpjulia — APPROVED** [review]
LGTM

**Comment by Simran-B** [comment]
https://jenkins.arangodb.biz:3456/view/PR/job/arangodb-matrix-pr/20370/

File: arangod/Agency/AgencyFeature.cpp [hunk 1]
```diff
@@ -154,13 +154,13 @@ void AgencyFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
...
           arangodb::options::Flags::OnAgent));
 
-  options->addOption(
-      "--agency.compaction-step-size",
-      "step size between state machine compactions",
-      new UInt64Parameter(&_compactionStepSize),
-      arangodb::options::makeFlags(
-          arangodb::options::Flags::DefaultNoComponents,
-          arangodb::options::Flags::Hidden, arangodb::options::Flags::OnAgent));
+  options->addOption("--agency.compaction-step-size",
```
[the remark this answers is not in the document]
Comment (Simran-B) [reply arangod/Agency/AgencyFeature.cpp:154]: As far as I can tell, this is what clang-format did. I reran it but don't see any changes.

**Pull request merged**